Crawler/SiteScraper/Scraper.cs

```diff
@@ -1,48 +1,66 @@
-﻿using Crawler.MockupWrappers;
+﻿using Crawler.Configs;
+using Crawler.MockupWrappers;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using Crawler.Exceptions;
 
-namespace Crawler
+namespace Crawler.SiteScraper
 {
-    public class ScienceDailyScraper : IScienceDailyScraper
+    public class Scraper : IScraper
     {
-        private const string SELECTOR = "div#text, p#first";
-
         private readonly IBrowsingContextWrapper context;
         private readonly ILogger logger;
+        private readonly ScrapersConfig config;
 
-        public ScienceDailyScraper(IBrowsingContextWrapper context, ILogger<ScienceDailyScraper> logger)
+        public Scraper(IBrowsingContextWrapper context, ILogger<Scraper> logger, IOptions<ScrapersConfig> config)
         {
             this.context = context;
             this.logger = logger;
+            this.config = config.Value;
         }
 
         public async Task<string> ScrapAsync(string address, CancellationToken cancellationToken)
         {
             var document = await context.OpenAsync(address, cancellationToken).ConfigureAwait(false);
             logger.LogDebug("Opened document from {address}", address);
 
-            var nodes = document.QuerySelectorAll(SELECTOR).ToList();
+            var selector = FindSelector(address);
+
+            if (selector is null)
+            {
+                logger.LogError("Could not find selector from config for {address}", address);
+                throw new SelectorNotFoundException($"Selector for site {address} not found");
+            }
+
+            var nodes = document.QuerySelectorAll(selector).ToList();
 
-            if (nodes.Count == 0)
+            if (!nodes.Any())
             {
-                logger.LogError("Could not find {selector} in {address}", SELECTOR, address);
+                logger.LogError("Could not find {selector} in {address}", selector, address);
 
                 return string.Empty;
             }
 
-            logger.LogDebug("Extracted {selector} from document", SELECTOR);
+            logger.LogDebug("Extracted {selector} from document", selector);
 
             var text = GetText(nodes);
 
-            logger.LogInformation("Extracted text from {address} {selector}", address, SELECTOR);
+            logger.LogInformation("Extracted text from {address} {selector}", address, selector);
 
             return text;
         }
 
+        private string? FindSelector(string url)
+        {
+            return config.ScrapesrDefinitions
+                .Select(scraperDefinition => scraperDefinition.Match(url))
+                .FirstOrDefault(selector => selector != null);
```

Comment:
Consider combining "Select" and "FirstOrDefault", although, it might be less readable.

Reply:
I don't see and advantage for combining it